### PR TITLE
Cloudwatch: Round up endTime in GetMetricData to next minute

### LINKED ISSUE #64568: Cloudwatch: last minute of metrics missing on small time-ranges
**What happened**:
When looking at the Cloudwatch Metrics datasource on a small period of time, such as a timerange of 2 minutes, the last minute of data is not returned.

**What you expected to happen**:
I expected the last datapoint to be there, if it existed in the aws console, but it is not showing up in the timeseries or in table view. The datapoint exists in the datasource and I can see it when I zoom out, but the last minute is always cut off.

**How to reproduce it (as minimally and precisely as possible)**:
- go to Cloudwatch Metrics in the cloudwatch datasource plugin on a recent version (I'm on main > 9.4, but I suspect this has been around for a while)
- create a sample metric, I selected this but I think any metric would be the same
Screenshot 2023-03-09 at 4 26 21 PM
- I then zoomed in to a small time range, I find 2 minutes and 45 seconds to be good for recreating this issue
- Notice how that time series seems to not have data after 12:01
Screenshot 2023-03-09 at 4 29 13 PM
- this is confirmed by the table where you see we have only 2 points at 12, 12:01, but 12:02 is missing
Screenshot 2023-03-09 at 4 31 16 PM
- however if I shift the time range to be 12-12:03 I can see we do have data for 12:02 but now it is 12:03 that is missing
Screenshot 2023-03-09 at 4 35 40 PM

When we look at the codebase, we see we're making a call to [GetMetricData](https://docs.aws.amazon.com/AmazonCloudWatch/latest/APIReference/API_GetMetricData.html) with an argument "endtime" which aws describes as: 
>The time stamp indicating the latest data to be returned.
>The value specified is exclusive; results include data points up to the specified time stamp.

It seems like Cloudwatch always rounds down the last minute and then considers that exclusive, which is causing us to have more graph than datapoints.

I think it might make sense to take the selected grafana time range, round it up to the next minute, and then pass that as endtime to Cloudwatch. If for some reason this is too wonky a solution, I think we should at least document this as I could see it being confusing for users.

### PR DESCRIPTION
**Pull request by idastambuk**
**Following @iwysiu's point, we agreed to add this as a feature toggle, so that users who query specific time ranges adjusted for their periods can disable it. Furthermore, we can disable rounding up if it causes other bugs.** 

**What is this feature?**

When looking at the Cloudwatch Metrics datasource on a small period of time, such as a time range of 2 minutes, the last minute of data is not returned.

From the original bug report: 

> When we look at the codebase, we see we're making a call to [GetMetricData](https://docs.aws.amazon.com/AmazonCloudWatch/latest/APIReference/API_GetMetricData.html) with an argument "endtime" which aws describes as:
> 
> The time stamp indicating the latest data to be returned.
> The value specified is exclusive; results include data points up to the specified time stamp.
> 
> It seems like Cloudwatch always rounds down the last minute and then considers that exclusive, which is causing us to have more graph than datapoints.

This rounds up the endTime field to the next minute before sending it to Cloudwatch

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/grafana/issues/64568

**Special notes for your reviewer:**

Please check that:
- [x] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/tsdb/cloudwatch/get_metric_data_executor_test.go [hunk 1]
```diff
@@ -12,9 +13,25 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
-func TestGetMetricDataExecutorTest(t *testing.T) {
+func TestGetMetricDataExecutorTestRequest_should_round_up_endTime(t *testing.T) {
```
Comment (sarahzinger) [inline pkg/tsdb/cloudwatch/get_metric_data_executor_test.go:16]: total nit and not important but why camelcase and snake case together? I think if you want an "it should do whatever" type pattern that we have with the jest tests I think you can do something like:
```
func TestGetMetricDataExcecutor(t *testing.T) {
    t.Run("should do things", func (t *testing.T) {
        // whatever the first test is
    }
    t.Run("should do other things", func (t *testing.T) {
        // whatever the second test is
    }
}
```

**Review by sarahzinger — APPROVED** [review]
LGTM

**Comment by iwysiu** [comment]
I'm ok with the change, but do you think we're going to have issues with this and with Cloudwatch's recommendation to align the start time and end time with the period for better performance? (mentioned at the bottom of the `Request Parameters` section of the [docs](https://docs.aws.amazon.com/AmazonCloudWatch/latest/APIReference/API_GetMetricData.html))

**Review by sarahzinger — APPROVED** [review]
Seems good! You have some merge conflicts and such, but once the build passes I'm cool to give this a try!

**Pull request merged**